possible_topics.md

```diff
@@ -108,6 +108,82 @@ The purpose of this document is to present the conventions and standards used at
 
 ### 3.4- Resolvers
 
+Resolvers, in GraphQL, work the same way for queries and mutations (nevertheless remember that we must not use a query resolver to modify nor delete data).
+
+- Given that the resolvers are functions, they must follow the naming convention used in the technology and project we are working in.
+
+- Resolvers should be kept as specific as possible. This way, we can take advantage of GraphQL performance.
+
+  For example. Lets suppose we have this query
+
+  ```graphql
+  query{
+    user(id: 1){
+      id
+      name
+      phones {
+        number
+        prefix
+      }
+    }
+  }
+  ```
+
+  If a resolver is used to get `id` and `name`, and another one is used to get the phones; when the user doesn't ask for the field phones, the resolver that get the phones won't be called.
+
+- Each resolver must return a key-value map (JSON), for all to produce the same structure than the user's query.
+
+  For example, this could be the response from the two resolvers:
+
+  ```json
+  {
+    "id": 1,
+    "name": "John Doe"
+  }
+  ```
+
+  ```json
+  [
+    {
+      "number": "12345678",
+      "prefix": "11"
+    },
+    {
+      "number": "87654321",
+      "prefix": "11"
+    }
+  ]
+  ```
+
+  And this the response generated by our router to user's query:
+
+  ```json
+  {
+    "data": {
+      "user": {
+        "id": 1,
+        "name": "John Doe",
+        "phones": [
+          {
+            "number": "12345678",
+            "prefix": "11"
+          },
+          {
+            "number": "87654321",
+            "prefix": "11"
+          },
+        ]
+      }
+    }
+  }
+  ```
+
+- Resolvers params (_<https://graphql.org> definition is used here_):
+  - `obj`: The previous object, which for a field on the root Query type is often not used.
+  - `args`: The arguments provided to the field in the GraphQL query.
+  - `context`: A value which is provided to every resolver and holds important contextual information like the currently logged in user, or access to a database.
+  - `info`: A value which holds field-specific information relevant to the current query as well as the schema details.
```

Comment:
Is this part of the GraphQL Spec (I can't seem to find it), or from a specific implementation like Apollo Server?

Reply:
It's part of gql specs. Here is the link: https://graphql.org/learn/execution/#root-fields-resolvers
I'll put that exact link instead of graphql.org :+1: